test/e2e/daemon_set.go

```diff
@@ -597,13 +597,25 @@ func checkDaemonPodsImage(c clientset.Interface, ns string, selector map[string]
 		}
 		pods := podList.Items
 
+		unavailablePods := 0
+		allImagesUpdated := true
 		for _, pod := range pods {
 			podImage := pod.Spec.Containers[0].Image
-			if podImage != image || !v1.IsPodReady(&pod) {
-				framework.Logf("Wrong image for pod: %s. Expected: %s, got: %s. Pod Ready: %t", pod.Name, image, podImage, v1.IsPodReady(&pod))
-				return false, nil
+			if podImage != image {
+				allImagesUpdated = false
+				framework.Logf("Wrong image for pod: %s. Expected: %s, got: %s.", pod.Name, image, podImage)
+			}
+			if !v1.IsPodAvailable(&pod, 0, metav1.Now()) {
```

Comment:
This is okay for now but is incorrect if DaemonSet's minReadySeond isn't zero. Maybe we should pass minReadySeond in.